ceph/iscsi/gateway.py

```diff
@@ -0,0 +1,183 @@
+        self.disks = self.Disks(self)
+        self.container = Container(self.node)
+        self._iscsi_container = get_iscsi_container(self.node)
+        self.container_id = self.iscsi_container["Names"][0]
+
+    @property
+    def iscsi_container(self):
+        return self._iscsi_container
+
+    @iscsi_container.setter
+    def iscsi_container(self, value):
+        self._iscsi_container = value
+        if not value:
+            self._iscsi_container = get_iscsi_container(self.node)
+
+    @staticmethod
+    def format_command_options(**kwargs):
+        """Construct the iscsi path to define target and its entities.
+
+        Args:
+            kwargs: command options in Dict
+
+        Returns:
+            Str
+        """
+        cmd_opts = str()
+        for k, v in kwargs.items():
+            cmd_opts += f" {k}={v}"
+        return cmd_opts
+
+    def exec_gw_cli(self, action, path, **kwargs):
+        """Execute iSCSI GWCLI command
+
+        Args:
+            action: operation to be performed on specific entity
+            path: iSCSI command path to manage entities
+            kwargs: command options
+        """
+        cmd = f"gwcli --debug {path} {action} {self.format_command_options(**kwargs)}"
+        return self.container.exec(
+            container=self.container_id,
+            interactive=True,
+            cmds=cmd,
+        )
+
+    def list(self, **kwargs):
+        return self.exec_gw_cli("ls", self.path, **kwargs)
+
+    class Disks:
+        path = "/disks"
+
+        def __init__(self, parent):
+            self.parent = parent
+
+        def attach(self, **kwargs):
+            return self.parent.exec_gw_cli("attach", self.path, **kwargs)
+
+        def create(self, **kwargs):
+            return self.parent.exec_gw_cli("create", self.path, **kwargs)
+
+        def delete(self, image_id):
+            return self.parent.exec_gw_cli(
+                "delete", self.path, **{"image_id": image_id}
+            )
+
+        def detach(self, image_id):
+            return self.parent.exec_gw_cli(
+                "detach", self.path, **{"image_id": image_id}
+            )
+
+        def list(self, path=""):
+            return self.parent.exec_gw_cli("list", self.path, **{"path": path})
+
+    class IQNTarget:
+        path = "/iscsi-targets"
+
+        def __init__(self, parent):
+            self.parent = parent
+            self.gateways = self.Gateways(parent)
+            self.hosts = self.Hosts(parent)
+            self.disks = self.Disks(parent)
+
+        def create(self, iqn):
+            return self.parent.exec_gw_cli("create", self.path, **{"target_iqn": iqn})
+
+        def delete(self, iqn):
+            return self.parent.exec_gw_cli("delete", self.path, **{"target_iqn": iqn})
+
+        def discovery_auth(self, **kwargs):
+            return self.parent.exec_gw_cli("discovery_auth", self.path, **kwargs)
+
+        class Disks:
+            path = "/iscsi-targets/{IQN}/disks"
```

Comment:
IMO, `path` here is an instance property as it is being dynamic.

Reply:
I'm thinking of it to initialize with IQN, But there are other paths in GWCLI like `/disks` and `/cluster` which doesn't need IQN. So to make it dynamic we may need different approach.

Reply:
I think this would not be possible, since we may have to initialize the GWCLI with IQN which is incorrect as per implementation and GWCLI paths.